### PR TITLE
feat: add basic layout

### LINKED ISSUE #1: Add basic layout
- display items
- navigation menu

### PR DESCRIPTION
**Pull request by pyphilia**
Most of the work was factoring out components from graasp-compose and use them from graasp-ui. 

Here some screenshots for a given "activity". 
- Text is small but you can edit it using the Editor
- iframes-links and pdfs will take most of the height by default
- The elements keep their original size, the maximum width is the screen width.
- You can navigate freely using the mainmenu as deep as defined when creating the folders. Using the mainmenu you can display one item at a time or view the content of 
- Files without preview (eg: zip): a button to download the file is displayed
- Child Folders: A button within the main screen is displayed. Clicking on it navigate to the folder.
- The Root folder can be any folder in your items.

This concerns more graasp-compose
- Static Apps such as Phet's can be added as a link or as an app.
- Images (and probably other files) can be added by uploading them or using links.

Screenshot 2021-06-09 at 17 01 39
Screenshot 2021-06-09 at 17 02 56
Screenshot 2021-06-09 at 17 03 04
Screenshot 2021-06-09 at 17 03 15
Screenshot 2021-06-09 at 17 03 22
Screenshot 2021-06-09 at 17 03 29
Screenshot 2021-06-09 at 17 03 41
Screenshot 2021-06-09 at 17 04 34
Screenshot 2021-06-09 at 17 04 49

close #1

### CODE REVIEW

File: cypress/support/utils.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  });
+  if (parseQueryString) {
+    characters.forEach((c) => {
+      querystring = querystring.replaceAll(c, `\\${c}`);
+    });
+  }
+  return `${pathname}${querystring.length ? '\\?' : ''}${querystring}`;
+};
+
+export const EMAIL_FORMAT = '[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\\.[a-zA-Z0-9-.]+';
```
Comment (juancarlosfarah) [inline cypress/support/utils.js:32]: Do we not have this in some repo so that we don't have to repeat it?

File: cypress/support/utils.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * Parse characters of a given string to return a correct regex string
+ * This function mainly allows for endpoints to have fixed chain of strings
+ * as well as regex descriptions for data validation, eg /items/item-login?parentId=<id>
+ *
+ * @param {string} string
+ * @param {string[]} characters
+ * @param {boolean} parseQueryString
+ * @returns regex string of the given string
+ */
+export const parseStringToRegExp = (
```
Comment (juancarlosfarah) [inline cypress/support/utils.js:14]: Same as below.

File: cypress/support/utils.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// use simple id format for tests
+export const ID_FORMAT = '(?=.*[0-9])(?=.*[a-zA-Z])([a-z0-9-]+)';
```
Comment (juancarlosfarah) [inline cypress/support/utils.js:2]: Same as below.

File: src/components/App.js [hunk 4]
```diff
@@ -5,14 +5,16 @@ import {
...
+import MainScreen from './main/MainScreen';
 
 export const App = () => (
   <Router>
     <Switch>
-      <Route path={HOME_PATH} exact component={Home} />
-      <Redirect to={HOME_PATH} />
+      <Route path="/" exact component={MainScreen} />
+      <Route path="/:rootId/:id" exact component={MainScreen} />
+      <Route path="/:rootId" exact component={MainScreen} />
```
Comment (juancarlosfarah) [inline src/components/App.js:16]: Why are the three the same screen?

File: src/components/common/Item.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+  });
+
+  // fetch file content if type is s3File
+  const { data: s3Content } = useS3FileContent(item?.get('id'), {
+    enabled: Boolean(item?.get('type') === ITEM_TYPES.S3_FILE),
+  });
+
+  // define a max height depending on the screen height
+  // use a bit less of the height because of the header and some margin
+  const maxHeight = window.innerHeight * 0.8;
```
Comment (juancarlosfarah) [inline src/components/common/Item.js:56]: Let's factor out this `0.8`.

File: src/enums/itemTypes.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  FOLDER: 'folder',
+  FILE: 'file',
+  S3_FILE: 's3File',
+  LINK: 'embeddedLink',
+  SHORTCUT: 'shortcut',
+  DOCUMENT: 'document',
+  APP: 'app',
+};
+
+Object.freeze(ITEM_TYPES);
```
Comment (juancarlosfarah) [inline src/enums/itemTypes.js:11]: 👍

File: src/utils/item.js [hunk 7]
```diff
@@ -0,0 +1,69 @@
+// eslint-disable-next-line no-useless-escape
```
Comment (juancarlosfarah) [inline src/utils/item.js:1]: Do you think it's worth it to factor this whole file out? I think that it's used almost identically in all of our front ends.

**Review by juancarlosfarah — CHANGES_REQUESTED** [review]
Very nice! Clean and compact. Just some minor questions, comments, and suggestions. There's still a question whether we should factor out utils to `graasp-utils` and/or `graasp-test` so as to avoid all the boilerplate. To be discussed.

File: src/components/App.js (hunk 4 above)
Comment (pyphilia) [reply src/components/App.js:16]: We have one screen only (yet) depending on the root item id and its children id. Currently `/` shows nothing (Item not defined screen). `/roodId` displays the root item's first level children, and `/rootId/id` selects directly the item, which displays this item's first level children.

**Comment by pyphilia** [comment]
As discussed, I think it would be ideal to have a `graasp-utils` and `graasp-test` , but for now it is faster for me to copy paste the code from one repo to the other... But it is in my plans. 
Having a repo for tests running on a dev instance of the client would be even greater and close to reality as it would include "real" loading time and use the real API.

**Pull request merged**